plugins/modules/storagebox_snapshot.py

```diff
@@ -0,0 +1,221 @@
+  returned: success
+  type: dict
+  contains:
+    name:
+      description: The name of the snapshot.
+      type: str
+      sample: "2025-01-21T12-40-38"
+    timestamp:
+      description: Timestamp of snapshot in UTC
+      type: str
+      sample: "2025-01-21T12:40:38+00:00"
+    size:
+      description: The size of the snapshot in MB.
+      type: int
+      sample: 400
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+from ansible_collections.community.hrobot.plugins.module_utils.robot import (
+    BASE_URL,
+    ROBOT_DEFAULT_ARGUMENT_SPEC,
+    fetch_url_json,
+)
+
+
+def handle_errors(module, error, storagebox_id=None, snapshot_name=None):
+    error_messages = {
+        "STORAGEBOX_NOT_FOUND": "Storagebox with ID {0} does not exist".format(storagebox_id),
+        "SNAPSHOT_NOT_FOUND": "Snapshot with name {0} does not exist".format(snapshot_name),
+        "SNAPSHOT_LIMIT_EXCEEDED": "Snapshot limit exceeded",
+    }
+    module.fail_json(msg=error_messages.get(error, error))
+
+
+def main():
+    argument_spec = dict(
+        storagebox_id=dict(type='int', required=True),
+        snapshot_name=dict(type='str'),
+        state=dict(type='str', default="present",
+                   choices=['present', 'absent']),
+        snapshot_comment=dict(type='str')
+    )
+
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_by={"snapshot_comment": ["snapshot_name"]},
+        required_if=[["state", "absent", ["snapshot_name"]]],
+        supports_check_mode=True
+    )
+
+    storagebox_id = module.params['storagebox_id']
+    state = module.params['state']
+    snapshot_name = module.params['snapshot_name']
+    snapshot_comment = module.params['snapshot_comment']
+
+    # Create snapshot
+    if state == 'present' and not snapshot_name:
+        if module.check_mode:
+            module.exit_json(changed=True)
+        snapshot = create_snapshot(module, storagebox_id)
+        module.exit_json(changed=True, snapshot=snapshot)
+
+    # Update snapshot comment
+    elif state == 'present' and snapshot_name:
+        if snapshot_comment is None:
+            module.fail_json(
+                msg="snapshot_comment is required when updating a snapshot")
+
+        snapshots = fetch_snapshots(
+            module=module, storagebox_id=storagebox_id)
+        snapshot = get_snapshot_by_name(snapshots, snapshot_name)
+        if not snapshot:
+            handle_errors(module, "SNAPSHOT_NOT_FOUND",
+                          snapshot_name=snapshot_name)
+        if snapshot_comment != snapshot['comment']:
+            if not module.check_mode:
+                update_snapshot_comment(
+                    module, storagebox_id, snapshot_name, snapshot_comment)
+            module.exit_json(changed=True)
+        else:
+            module.exit_json(changed=False)
```

Comment:
I'd also return the snapshot in this case:
```suggestion
            module.exit_json(changed=True, snapshot=snapshot)
        else:
            module.exit_json(changed=False, snapshot=snapshot)
```